src/App.tsx

```diff
@@ -26,70 +27,134 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((el1, el2) => el1.localeCompare(el2));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((el1, el2) => el1.length - el2.length);
+      break;
+    default:
+      visibleGoods.map(el => el);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleSortAlphabetic = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+    });
+  };
+
+  handleSortLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+  };
+
+  handleSortReverse = () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+    this.state.isReversed ? this.setState({
+      isReversed: false,
+    }) : this.setState({
+      isReversed: true,
+    });
```

Comment:
You could just change it to opposite this way 👇 
```suggestion
    this.setState(state => ({
      isReversed: !state.isReversed,
    }));
```
And `eslint-disable` this way would be redundant, I think.